puppet/custom_device_context.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+
+import argparse
+import json
+import requests
+
+from requests.auth import HTTPBasicAuth
+from typing import List
+
+SOURCE = "puppet"
+
+
+def read_data_file(file: str) -> List:
+    f = open(file)
+    data = json.load(f)
+    f.close()
+    return data
```

Comment:
Instead of just loading raw data, use a Dataclass so we know the data is as per the JSON structure we expect.